### PR TITLE
dev: add CapsuleTest trait to test doc

### PR DESCRIPTION
**Pull request by bradjc**
Include `CapsuleTest` in doc on writing kernel tests.

### CODE REVIEW

File: src/development/tests.md [hunk 1]
```diff
@@ -124,75 +123,86 @@ The steps from the overview are elaborated on here.
...
+        assert!(result.is_err());
+        socket1 = result.unwrap_err(); // Get the socket back
+
+        //now bind to an open port
+        let (_send_bind, _recv_bind) = self
+            .port_table
+            .bind(socket1, 1001, net_cap)
+            .expect("UDP Bind fail");
+
+        debug!("bind_test passed");
```
Comment (brghena) [inline src/development/tests.md:198]: Shouldn't this be calling

```
self.client.map(|client| {
  client.done(Ok(()));
});
```

Comment (brghena) [reply src/development/tests.md:198]: Or are you intending to save that for the next part of the tutorial?

File: src/development/tests.md [hunk 2]
```diff
@@ -242,22 +260,18 @@ The steps from the overview are elaborated on here.
...
            let (err, _opt) = self.device.receive_buffer(rx_buffer, rx_len);
            if err != ReturnCode::SUCCESS {
-               panic!(
+               debug!(
                    "Calling receive_buffer() in virtual_uart test failed: {:?}",
                    err
                );
+               self.client.map(|client| {
+                   client.done(Err(CapsuleTestError::ErrorCode(ErrorCode::FAIL)));
+               });
```
Comment (brghena) [inline src/development/tests.md:269]: Shouldn't this also send an `Ok()` if it wasn't an error?

**Review by brghena — COMMENTED** [review]
Two related questions about sending an OK if the test passes. Otherwise this is ready to go.

File: src/development/tests.md (hunk 1 above)
Comment (bradjc) [reply src/development/tests.md:198]: Umm, not sure. What do you think?

Comment (brghena) [reply src/development/tests.md:198]: Well, you set a client. So sending an OK seems reasonable to me.

File: src/development/tests.md (hunk 2 above)
Comment (bradjc) [reply src/development/tests.md:273]: Added the OK case

**Review by ppannuto — APPROVED** [review]

**Pull request merged**